src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -62,85 +122,111 @@ int FilterParserThread::parseDATFilterFile()
+                // but first move(copy) the leftover data to the front of the buffer
+                offset = dataSize - start;
+                memmove(buffer, buffer + start, offset);
+                break;
+            }
+            else {
+                ++nbLine;
+            }
 
-        // IP Range should be split by a dash
-        QList<QByteArray> IPs = partsList.first().split('-');
-        if (IPs.size() != 2) {
-            qDebug("Ipfilter.dat: line %d is malformed.", nbLine);
-            qDebug("Line was %s", line.constData());
-            continue;
-        }
+            if ((buffer[start] == '#')
+                || ((buffer[start] == '/') && ((start + 1 < dataSize) && (buffer[start + 1] == '/')))) {
+                start = endOfLine;
+                continue;
+            }
 
-        boost::system::error_code ec;
-        const QString strStartIP = cleanupIPAddress(IPs.at(0));
-        if (strStartIP.isEmpty()) {
-            qDebug("Ipfilter.dat: line %d is malformed.", nbLine);
-            qDebug("Start IP of the range is malformated: %s", qPrintable(strStartIP));
-            continue;
-        }
-        libt::address startAddr = libt::address::from_string(qPrintable(strStartIP), ec);
-        if (ec) {
-            qDebug("Ipfilter.dat: line %d is malformed.", nbLine);
-            qDebug("Start IP of the range is malformated: %s", qPrintable(strStartIP));
-            continue;
-        }
+            QVector<int> delimIndices = indicesOfDelimiters(buffer, ',', start, endOfLine);
+
+            // Check if there is an access value (apparently not mandatory)
+            if (!delimIndices.isEmpty()) {
+                // There is possibly one
+                int sizeOfAccessCode = (delimIndices.size() > 1
+                                        ? ((delimIndices[1] - 1) - (delimIndices[0] + 1) + 1) : ((endOfLine - 1) - (delimIndices[0] + 1) + 1));
+                const QByteArray accesscode(buffer + (delimIndices[0] + 1), sizeOfAccessCode);
+                const int nbAccess = accesscode.toInt();
+                // Ignoring this rule because access value is too high
+                if (nbAccess > 127) {
+                    start = endOfLine;
+                    continue;
+                }
+            }
 
-        const QString strEndIP = cleanupIPAddress(IPs.at(1));
-        if (strEndIP.isEmpty()) {
-            qDebug("Ipfilter.dat: line %d is malformed.", nbLine);
-            qDebug("End IP of the range is malformated: %s", qPrintable(strEndIP));
-            continue;
-        }
+            // IP Range should be split by a dash
+            int endOfIPRange = (delimIndices.isEmpty() ? (endOfLine - 1) : (delimIndices[0] - 1));
+            QVector<int> delimIP = indicesOfDelimiters(buffer, '-', start, endOfIPRange);
+            if (delimIP.size() != 1) {
+                Logger::instance()->addMessage(tr("IP filter line %1 is malformed.").arg(nbLine), Log::CRITICAL);
```

Comment:
I'm very late with this, of course, but wouldn't `Log(type, message)` be a better signature?